include/lsst/afw/detection/Footprint.h

```diff
@@ -340,212 +235,19 @@ class Footprint : public lsst::daf::base::Citizen,
 };
 
-/**
- Given a vector of Footprints, fills the output "argmin" and "dist"
- images to contain the Manhattan distance to the nearest footprint (in
- "dist") and the identity of the nearest footprint (in "argmin").
-
- For example, if there are two footprints at y=0 covering x=[1,2] and [7,7],
-
- Index     :  0 1 2 3 4 5 6 7
-
- Footprints:  . 0 0 . . . . 1
-
- Argmin    :  0 0 0 0 0 1 1 1
-
- Dist      :  1 0 0 1 2 2 1 0
-
- "argmin" gives the index of the nearest footprint, and "dist" its
- Manhattan (L_1 norm) distance.  The pixel at index 4 is closest to
- footprint 0, and its distance is 2.
- */
-void nearestFootprint(std::vector<PTR(Footprint)> const& foots,
-                      lsst::afw::image::Image<std::uint16_t>::Ptr argmin,
-                      lsst::afw::image::Image<std::uint16_t>::Ptr dist);
-
-/**
-   Merges two Footprints -- appends their peaks, and unions their
-   spans, returning a new Footprint.
-
-   This const version requires that both input footprints are
-   normalized (and will raise an exception if not).
- */
-PTR(Footprint) mergeFootprints(Footprint const& foot1, Footprint const& foot2);
-
-/**
-   Merges two Footprints -- appends their peaks, and unions their
-   spans, returning a new Footprint.
- */
-PTR(Footprint) mergeFootprints(Footprint& foot1, Footprint& foot2);
-
-/**
- * Shrink a footprint isotropically by nGrow pixels, returning a new Footprint.
- */
-PTR(Footprint) shrinkFootprint(Footprint const& foot, int nGrow, bool isotropic);
-
-/**
- * Grow a Footprint by nGrow pixels, returning a new Footprint.
- */
-PTR(Footprint) growFootprint(Footprint const& foot, int nGrow, bool isotropic=true);
-
-/**
- * \note Deprecated interface; use the Footprint const& version.
- */
-PTR(Footprint) growFootprint(PTR(Footprint) const& foot, int nGrow, bool isotropic=true);
-
-/**
- * \brief Grow a Footprint in at least one of the cardinal directions,
- * returning a new Footprint
- *
- * Note that any left/right grow is done prior to the up/down grow, so
- * any left/right grown pixels \em are subject to a further up/down
- * grow (i.e. an initial single pixel Footprint will end up as a
- * square, not a cross.
- */
-PTR(Footprint) growFootprint(Footprint const& foot, int nGrow,
-                             bool left, bool right, bool up, bool down);
-
-/**
- * Return a list of BBox%s, whose union contains exactly the pixels in
- * foot, neither more nor less
- *
- * Useful in generating sets of meas::algorithms::Defects for the ISR
- */
-std::vector<lsst::afw::geom::Box2I> footprintToBBoxList(Footprint const& foot);
-
-/**
- * \brief Set all image pixels in a Footprint to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprint(ImageT *image,
-                                             Footprint const& footprint,
-                                             typename ImageT::Pixel const value);
```

Comment:
While it's now trivial to do this in C++ to do this with `SpanSet::applyFunctor` and a lambda, we definitely need to preserve the ability to do it from Python.  However, it should probably be moved to `SpanSet`, and it should take the `Image` argument by reference instead of by pointer.  I also wouldn't object to only defining it in the Python interface.